cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -60,20 +60,65 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
     endif()
   else()
-    find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED QUIET CONFIG PATHS
-                 /usr
-                 /usr/local
-                 /opt
-                 $ENV{HOME}
-                 $ENV{HOME}/.local
-                 $ENV{HOME}/.local/opt
-                 $ENV{HOME}/bin)
+    # Paths that are used to find installed Zephyr SDK versions
+    SET(ZEPHYR_SDK_SEARCH_PATHS
+        /usr
+        /usr/local
+        /opt
+        $ENV{HOME}
+        $ENV{HOME}/.local
+        $ENV{HOME}/.local/opt
+        $ENV{HOME}/bin)
+
+    # Search for Zephyr SDK version 0.0.0 which does not exist, this is needed to
+    # return a list of compatible versions and find the best suited version that
+    # is available.
+    find_package(Zephyr-sdk 0.0.0 EXACT QUIET CONFIG PATHS ${ZEPHYR_SDK_SEARCH_PATHS})
+
+    # Remove duplicate entries and sort naturally in descending order.
+    set(ZEPHYR_SDK_FOUND_VERSIONS ${Zephyr-sdk_CONSIDERED_VERSIONS})
+    set(ZEPHYR_SDK_FOUND_CONFIGS ${Zephyr-sdk_CONSIDERED_CONFIGS})
+
+    list(REMOVE_DUPLICATES Zephyr-sdk_CONSIDERED_VERSIONS)
+    list(SORT Zephyr-sdk_CONSIDERED_VERSIONS COMPARE NATURAL ORDER DESCENDING)
+
+    # Loop over each found Zepher SDK version until one is found that is compatible.
+    foreach(ZEPHYR_SDK_CANDIDATE ${Zephyr-sdk_CONSIDERED_VERSIONS})
+      if("${ZEPHYR_SDK_CANDIDATE}" VERSION_GREATER_EQUAL "${Zephyr-sdk_FIND_VERSION}")
+        # Find the path for the current version being checked and get the directory
+        # of the Zephyr SDK so it can be checked.
+        list(FIND ZEPHYR_SDK_FOUND_VERSIONS ${ZEPHYR_SDK_CANDIDATE} ZEPHYR_SDK_CURRENT_INDEX)
+        list(GET ZEPHYR_SDK_FOUND_CONFIGS ${ZEPHYR_SDK_CURRENT_INDEX} ZEPHYR_SDK_CURRENT_CHECK_PATH)
+        get_filename_component(ZEPHYR_SDK_CURRENT_CHECK_PATH ${ZEPHYR_SDK_CURRENT_CHECK_PATH} DIRECTORY)
+
+        # Then see if this version is compatible.
+        find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} QUIET CONFIG PATHS ${ZEPHYR_SDK_CURRENT_CHECK_PATH} NO_DEFAULT_PATH)
+
+        if (${Zephyr-sdk_FOUND})
+          # A compatible version of the Zephyr SDK has been found which is the highest
+          # supported version, exit.
+          break()
+        endif()
+      endif()
+    endforeach()
+
+    if (NOT ${Zephyr-sdk_FOUND})
+      # This means no compatible Zephyr SDK versions were found, set the version
+      # back to the minimum version so that it is displayed in the error text.
+      find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED CONFIG PATHS ${ZEPHYR_SDK_SEARCH_PATHS})
+    endif()
   endif()
 
   SET(CMAKE_FIND_PACKAGE_SORT_DIRECTION ${CMAKE_FIND_PACKAGE_SORT_DIRECTION_CURRENT})
   SET(CMAKE_FIND_PACKAGE_SORT_ORDER ${CMAKE_FIND_PACKAGE_SORT_ORDER_CURRENT})
 endif()
 
+# Clean up temp variables
+set(ZEPHYR_SDK_FOUND_VERSIONS)
+set(ZEPHYR_SDK_FOUND_CONFIGS)
+set(ZEPHYR_SDK_CURRENT_INDEX)
+set(ZEPHYR_SDK_CURRENT_CHECK_PATH)
```

Comment:
we generally use lower case for internal vars to make it easier to distinguish those from global variables.

Reply:
Fixed